.github/workflows/publish.yml

```diff
@@ -14,16 +14,17 @@ jobs:
       matrix:
         java: ['11']
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
         with:
           path: ~/.gradle/caches
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
       - name: Set up JDK
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v4
         with:
+          distribution: 'adopt'
           java-version: ${{ matrix.java }}
       - name: Publish Main Site
         run: ./publish.sh
```

Comment:
As `./publish.sh` invokes `./gradlew` we should probably do `gradle/actions/setup-gradle@v4` as a step as well. And also pass the build cache environment variables.